// actions/actionTypes.js
//добавить зеркальную библиотеку
//при загрузке ноутбуков на страницу
export const FETCH_NOTEBOOKS_START = 'FETCH_NOTEBOOKS_START'
export const FETCH_NOTEBOOKS_SUCCESS = 'FETCH_NOTEBOOKS_SUCCESS'
export const FETCH_NOTEBOOKS_FAILURE = 'FETCH_NOTEBOOKS_FAILURE'

//при нажатии на кнопку "показать больше"
export const LOAD_MORE_NOTEBOOKS_START = 'LOAD_MORE_NOTEBOOKS_START'
export const LOAD_MORE_NOTEBOOKS_SUCCESS = 'LOAD_MORE_NOTEBOOKS_SUCCESS'
export const LOAD_MORE_NOTEBOOKS_FAILURE = 'LOAD_MORE_NOTEBOOKS_FAILURE'

//при нажатии на кнопку ноутбука попадаем на отдельную страницу
export const FETCH_NOTEBOOK_BY_ID_START = 'FETCH_NOTEBOOK_BY_ID_START'
export const FETCH_NOTEBOOK_BY_ID_SUCCESS = 'FETCH_NOTEBOOK_BY_ID_SUCCESS'
export const FETCH_NOTEBOOK_BY_ID_FAILURE = 'FETCH_NOTEBOOK_BY_ID_FAILURE'

//загрузка категорий
export const FETCH_CATEGORIES_START = 'FETCH_CATEGORIES_START'
export const FETCH_CATEGORIES_SUCCESS = 'FETCH_CATEGORIES_SUCCESS'
export const FETCH_CATEGORIES_FAILURE = 'FETCH_CATEGORIES_FAILURE'

//так как это синхронный оператор, то Action будет один
//для поиска ноутбука
export const SEARCH_NOTEBOOK = 'SEARCH_NOTEBOOK'
